src/Facility.Definition/ServiceFieldValidationRange.cs

```diff
@@ -0,0 +1,83 @@
+using System;
+
+namespace Facility.Definition
+{
+	/// <summary>
+	/// A range for validating numeric values
+	/// </summary>
+	public sealed class ServiceFieldValidationRange
+	{
+		/// <summary>
+		/// Creates a range for validation
+		/// </summary>
+		/// <param name="startInclusive">The inclusive start of the range</param>
+		/// <param name="endInclusive">The inclusive end of the range</param>
+		public ServiceFieldValidationRange(decimal? startInclusive, decimal? endInclusive)
+		{
+			if (startInclusive == null && endInclusive == null)
+				throw new ArgumentException($"{nameof(startInclusive)} or {nameof(endInclusive)} must be specified");
+
+			StartInclusive = startInclusive;
+			EndInclusive = endInclusive;
+		}
+
+		/// <summary>
+		/// The start of the range
+		/// </summary>
+		public decimal? StartInclusive { get; }
+
+		/// <summary>
+		/// The end of the range
+		/// </summary>
+		public decimal? EndInclusive { get; }
+
+		/// <summary>
+		/// Verifies if a value falls within the range
+		/// </summary>
+		/// <param name="value">The value</param>
+		/// <returns><c>true</c> if the range contains the value</returns>
+		public bool Contains(decimal value) => !(StartInclusive > value || EndInclusive < value);
```

Comment:
I don't see how `Contains` will ever be used. This code is not used by clients or servers validating data, but by code generators that will generate the code that validates the values.